core/embed/projects/prodtest/cmd/prodtest_secrets.c

```diff
@@ -0,0 +1,136 @@
+/*
+ * This file is part of the Trezor project, https://trezor.io/
+ *
+ * Copyright (c) SatoshiLabs
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <trezor_model.h>
+
+#include <string.h>
+
+#include <rtl/cli.h>
+#ifdef USE_OPTIGA
+#include <sec/optiga_commands.h>
+#endif
+#include <sec/secret.h>
+#include <sec/secret_keys.h>
+
+#include "memzero.h"
+#include "rand.h"
+#include "secbool.h"
+
+secbool generate_random_secret(uint8_t* secret, size_t length) {
+  random_buffer(secret, length);
+
+#ifdef USE_OPTIGA
+  uint8_t optiga_secret[length];
+  if (OPTIGA_SUCCESS != optiga_get_random(optiga_secret, length)) {
+    return secfalse;
+  }
+  for (size_t i = 0; i < length; i++) {
+    secret[i] ^= optiga_secret[i];
+  }
+  memzero(optiga_secret, sizeof(optiga_secret));
+#endif
+
+#ifdef USE_TROPIC
+  // TODO: Generate randomness using tropic and xor it with `secret`.
+#endif
+
+  return sectrue;
+}
+
+secbool set_random_secret(uint8_t slot, size_t length) {
+  uint8_t secret[length];
+  uint8_t secret_read[length];
+
+  secbool ret = secfalse;
+
+  if (secret_key_get(slot, secret_read, sizeof(secret_read)) != sectrue) {
```

Comment:
`secret_key_writable` would be better , otherwise you will fail with RSOD when this is attempted on unlocked bootloader (`secret_key_get` returns `secfalse` as the key is no longer present, but it cannot be written as the flash is locked)